src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2761,12 +2761,8 @@ private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression oper
                     }
                 case BoundKind.UnconvertedCollectionExpression:
                     {
-                        if (operand.Type is null)
-                        {
-                            Error(diagnostics, ErrorCode.ERR_CollectionExpressionTargetTypeNotConstructible, syntax, targetType);
-                            return;
-                        }
-                        break;
+                        GenerateImplicitConversionErrorForCollectionExpression((BoundUnconvertedCollectionExpression)operand, targetType, diagnostics);
```

Comment:
Should the method be renamed since it is used to handle both implicit and explicit conversions now?

Reply:
I'd considered renaming but decided to leave the name as is since we're using a similarly named method for the UnconvertedConditionalOperator and UnconvertedSwitchExpression cases above.

Perhaps we can consider renaming in a follow up PR if necessary.